types/backbone-relational/index.d.ts

```diff
@@ -7,166 +7,170 @@
+        setKeyContents(keyContents:number):void;
 
-    setKeyContents(keyContents:BModel):void;
+        setKeyContents(keyContents:Model):void;
 
-    onChange(model:BModel, attr:any, options:any):void;
+        onChange(model:Model, attr:any, options:any):void;
 
-    tryAddRelated(model:BModel, coll:any, options:any):void;
+        tryAddRelated(model:Model, coll:any, options:any):void;
 
-    addRelated(model:BModel, options:any):void;
+        addRelated(model:Model, options:any):void;
 
-    removeRelated(model:BModel, coll:any, options:any):void;
+        removeRelated(model:Model, coll:any, options:any):void;
 
-}
+    }
 
-export class Store extends EventsMixin implements Events {
-    initializeRelation(model, relation, options);
+    class Store extends EventsMixin implements Events {
+        initializeRelation(model, relation, options);
 
-    addModelScope(scope:any):void;
+        addModelScope(scope:any):void;
 
-    removeModelScope(scope):void;
+        removeModelScope(scope):void;
 
-    addSubModels(subModelTypes:Model, superModelType:Model):void;
+        addSubModels(subModelTypes:RelationalModel, superModelType:RelationalModel):void;
 
-    setupSuperModel(modelType:Model):void;
+        setupSuperModel(modelType:RelationalModel):void;
 
-    addReverseRelation(relation:any):void;
+        addReverseRelation(relation:any):void;
 
-    addOrphanRelation(relation:any):void;
+        addOrphanRelation(relation:any):void;
 
-    processOrphanRelations():void;
+        processOrphanRelations():void;
 
-    retroFitRelation(relation:Model, create:boolean):Collection<BModel>;
+        retroFitRelation(relation:RelationalModel, create:boolean):Collection;
 
-    getCollection(type:Model, create:boolean):Collection<BModel>;
+        getCollection(type:RelationalModel, create:boolean):Collection;
 
-    getObjectByName(name:string):any;
+        getObjectByName(name:string):any;
 
 
-    resolveIdForItem(type:any, item:any):any;
+        resolveIdForItem(type:any, item:any):any;
 
-    static find(type:any, item:string):Model;
+        static find(type:any, item:string):RelationalModel;
 
-    static find(type:any, item:number):Model;
+        static find(type:any, item:number):RelationalModel;
 
-    static find(type:any, item:Model):Model;
+        static find(type:any, item:RelationalModel):RelationalModel;
 
-    static find(type:any, item:any):Model;
+        static find(type:any, item:any):RelationalModel;
 
-    register(model:Model):void;
+        register(model:RelationalModel):void;
 
-    checkId(model:Model, id:any):void;
+        checkId(model:RelationalModel, id:any):void;
 
-    update(model:Model):void;
+        update(model:RelationalModel):void;
 
-    // tslint:disable-next-line use-default-type-parameter
-    unregister(type: Model | Collection<Model> | typeof Model): void;
+        // tslint:disable-next-line use-default-type-parameter
+        unregister(type: Model | Collection | typeof Model): void;
```

Comment:
You just missed updating these type references.

```suggestion
        unregister(type: RelationalModel | Collection | typeof RelationalModel): void;
```

Strictly speaking, the error is legit (I don’t understand the library enough to know whether it represents a “real” unsoundness, but the assignability error makes sense in theory), so the question is not “why is this erroring” but “why is it _not_ erroring in `master`.” The answer is that in `master`, one of these union constituents is `typeof RelationalModel` which has the type `any` for the property `relations`. So while `typeof Backbone.Model` (and its subclass constructor types) are all invariant due to referencing `this` in both covariant and contravariant positions, `typeof Person` _is_ assignable to `typeof Backbone.RelationalModel` because the latter provides an `any` escape hatch for `Person`’s only property that makes it structurally different from its base class.

Reply:
Thanks @andrewchang! I have no idea how I could have missed this, since (I think) I used a search-replace in order to update everything in one sweep. At least that explains why I didn't think of this.